File: PolishPizza/admin/src/service/drinkService.ts
import { api as axiosInstance } from '../api/api';
import type { DrinkResponse, DrinkRequest } from '../types/menu';

export const DrinkService = {
    
    getAll: async (): Promise<DrinkResponse[]> => {
        const response = await axiosInstance.get<DrinkResponse[]>('/menu/drinks'); 
        return response.data;
    },

    getOne: async (id: string): Promise<DrinkResponse | null> => {
        const response = await axiosInstance.get<DrinkResponse>(`/menu/drinks/${id}`);
        return response.data;
    },
    
    update: async (id: string, data: DrinkRequest) => {
        const response = await axiosInstance.put(`/menu/drinks/${id}`, data);
        return response.data;
    },
    
    create: async (data: DrinkRequest) => {
        const response = await axiosInstance.post(`/menu/drinks`, data);
        return response.data;
    },

    delete: async(id: string): Promise<void> => {
        const response = await axiosInstance.delete(`/menu/drinks/${id}`);
        return response.data;
    }
};

File: PolishPizza/menu/src/main/java/pizzeria/menu/drink/service/DrinkService.java
package pizzeria.menu.drink.service;

import pizzeria.menu.drink.dto.response.DrinkResponse;
import pizzeria.menu.drink.dto.request.DrinkPatchRequest;
import pizzeria.menu.drink.dto.request.DrinkRequest;

import java.util.List;
import java.util.UUID;

public interface DrinkService {
    List<DrinkResponse> getAllDrinks();

    DrinkResponse getDrinkById(UUID id);

    DrinkResponse save(DrinkRequest request);

    void delete(UUID id);

    DrinkResponse update(UUID id, DrinkRequest request);

    DrinkResponse patch(UUID id, DrinkPatchRequest request);
}
